modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManager.java

```diff
@@ -1,424 +0,0 @@
-/*
- * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.wso2.charon3.core.protocol.endpoints;
-
-import org.wso2.charon3.core.encoder.JSONDecoder;
-import org.wso2.charon3.core.encoder.JSONEncoder;
-import org.wso2.charon3.core.exceptions.BadRequestException;
-import org.wso2.charon3.core.exceptions.CharonException;
-import org.wso2.charon3.core.exceptions.ConflictException;
-import org.wso2.charon3.core.exceptions.InternalErrorException;
-import org.wso2.charon3.core.exceptions.NotFoundException;
-import org.wso2.charon3.core.exceptions.NotImplementedException;
-import org.wso2.charon3.core.extensions.UserManager;
-import org.wso2.charon3.core.objects.User;
-import org.wso2.charon3.core.protocol.ResponseCodeConstants;
-import org.wso2.charon3.core.protocol.SCIMResponse;
-import org.wso2.charon3.core.schema.SCIMConstants;
-import org.wso2.charon3.core.schema.SCIMResourceSchemaManager;
-import org.wso2.charon3.core.schema.SCIMResourceTypeSchema;
-import org.wso2.charon3.core.schema.ServerSideValidator;
-import org.wso2.charon3.core.utils.CopyUtil;
-import org.wso2.charon3.core.utils.PatchOperationUtil;
-import org.wso2.charon3.core.utils.ResourceManagerUtil;
-import org.wso2.charon3.core.utils.codeutils.PatchOperation;
-
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-/**
- * A client MAY use a URL of the form "<base-uri>/Me" as a uri alias for
- * the User or other resource associated with the currently
- * authenticated subject for any SCIM operation.
- */
-
-public class MeResourceManager extends AbstractResourceManager {
```

Comment:
This is a main class in Charon repo. Removing public classes are not recommended.  There may be users who use this class. Changes should be done in a backward-compatible manner.